File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    if(!head || !head.next) return head;
    let dummy=new ListNode();
    dummy.next=head;
    let prev = dummy;
    let curr=head;
    let nextNode= head.next;
    while(curr && nextNode){
        prev.next=nextNode;
        curr.next=nextNode.next;
        nextNode.next=curr;
        prev=curr;
        curr=prev.next;
        nextNode=curr && curr.next;
    }
    return dummy.next;
    
};

File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy=new ListNode(0);
        dummy.next=head;
        ListNode prev=dummy;
        while(prev.next!=null && prev.next.next!=null){
            ListNode curr=prev.next;
            ListNode n=curr.next;
            curr.next=n.next;
            n.next=curr;
            prev.next=n;
            prev=curr;
        }
        return dummy.next;
    }
}